src/plugins/discover/public/application/components/context_app/context_app_legacy.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  DocTableLegacy,
+  DocTableLegacyProps,
+} from '../../angular/doc_table/create_doc_table_react';
+import { IIndexPattern, IndexPatternField } from '../../../../../data/common/index_patterns';
+import { LOADING_STATUS } from './constants';
+
+export interface ContextAppProps {
+  columns: string[];
+  hits: Array<Record<string, unknown>>;
+  indexPattern: IIndexPattern;
+  filter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
+  minimumVisibleRows: number;
+  sorting: string[];
+  status: string;
+}
+
+export function ContextAppLegacy(renderProps: ContextAppProps) {
+  const { hits, filter, sorting, status } = renderProps;
+  const props = ({ ...renderProps } as unknown) as DocTableLegacyProps;
+  props.rows = hits;
+  props.onFilter = filter;
+  props.sort = sorting.map((el) => [el]);
```

Comment:
with the following code

 ```
const { hits, filter, sorting, status, ...rest } = renderProps;
  const props = {
    ...rest,
    rows: hits,
    onFilter: filter,
    sort: sorting.map((el) => [el]),
  };
```
`unknown` and `DocTableLegacyProps` would be redundant

Reply:
I'll fix this in the follow-up PR, when I'll be moving props around to accommodate for `context-action-bar` component and its props.